### ISSUE TITLE
`device_map="auto"` doesn't use all available GPUs when `load_in_8bit=True`

### ISSUE THREAD

**Issue by yukw777** [huggingface/transformers#22595]
### System Info

- `transformers` version: 4.28.0.dev0
- Platform: Linux-4.18.0-305.65.1.el8_4.x86_64-x86_64-with-glibc2.28
- Python version: 3.10.4
- Huggingface_hub version: 0.13.3
- PyTorch version (GPU?): 2.0.0+cu117 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: yes
- Using distributed or parallel set-up in script?: yes

### Who can help?

@sgugger @kooshi 

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

On a machine with more than 2 GPUs (I have 4*A40s)
```python
model = transformers.LlamaForCausalLM.from_pretrained(
    "path/to/converted/llama-65B",
    load_in_8bit=True,
    device_map="auto"
)
```
You'll see that only the first two GPUs are filled up. Possibly related to https://github.com/huggingface/transformers/pull/22377.

### Expected behavior

All 4 GPUs should get parameters.

**Comment by yukw777** [comment]
If I specify `max_memory`, the parameters do get distributed according to it.

```python
model = transformers.LlamaForCausalLM.from_pretrained(
    "path/to/converted/llama-65B",
    load_in_8bit=True,
    device_map="auto",
    max_memory={0: "10GB", 1: "10GB", 2: "48GB", 3: "48GB"}
)
```

**Comment by sgugger** [comment]
cc @younesbelkada

**Comment by Xmaster6y** [comment]
You can also specify `device_map="balanced"` to get the parameters evenly dispatched.

**Comment by younesbelkada** [comment]
Hmm maybe this is unrelated to `load_in_8bit`, can you try without that and let us know?
on the other hand I second what @Xmaster6y said, you can use `balanced` in this case

**Comment by sgugger** [comment]
balanced and auto are the same thing FYI.

**Comment by yukw777** [comment]
Yup, I knew `auto` and `balanced` were the same, but tried `balanced` for good measure. Same behavior :/.

I just verified that without `load_in_8bit`, the parameters are distributed evenly among the GPUs as expected.

**Comment by sgugger** [comment]
So this is specifically for `load_in_8bit`. I think @younesbelkada made a fix after the last patch. Could you try an install from source?

**Comment by yukw777** [comment]
I already installed from source.

```
$ pip freeze | grep transformers
transformers @ git+https://github.com/huggingface/transformers.git@15641892985b1d77acc74c9065c332cd7c3f7d7f
```

**Comment by kooshi** [comment]
I can't dig too deeply into this until later, and I don't have more than 2 GPUs to test, but I can say that the actual size calculations and dispatch are all done in [accelerate](https://github.com/huggingface/accelerate), and the calculation changed as little as 3 weeks ago, so make sure you have the latest installed.

If that doesn't fix it, and you want to dig into it, I'd recommend just sprinkling some `print`s around the relevant `accelerate` and `transformers` functions, just to get some visibility into what it thinks it's calculating.

It'll be somewhere in
`transformers/modeling_utils.py` which calls
`get_balanced_memory` and `infer_auto_device_map` in `accelerate/utils/modeling.py`

**Comment by yukw777** [comment]
Thanks @kooshi! It seems like something was fixed since accelerate 0.8. Installing accelerate from source resolved this issue.

**Comment by mclassen** [comment]
I'm still running into this
